src/Discord.Net.WebSocket/Entities/Interaction/Message Components/SocketMessageComponent.cs

```diff
@@ -109,45 +109,104 @@ internal override void Update(Model model)
+
+            if (args.AllowedMentions.IsSpecified)
+            {
+               var allowedMentions = args.AllowedMentions.Value;
+               Preconditions.AtMost(allowedMentions?.RoleIds?.Count ?? 0, 100, nameof(allowedMentions), "A max of 100 role Ids are allowed.");
+               Preconditions.AtMost(allowedMentions?.UserIds?.Count ?? 0, 100, nameof(allowedMentions), "A max of 100 user Ids are allowed.");
+            }
+
+            if (args.Embeds.IsSpecified)
+                Preconditions.AtMost(args.Embeds.Value?.Length ?? 0, 10, nameof(args.Embeds), "A max of 10 embeds are allowed.");
+
+            // check that user flag and user Id list are exclusive, same with role flag and role Id list
+            if (args.AllowedMentions.IsSpecified && args.AllowedMentions.Value != null && args.AllowedMentions.Value.AllowedTypes.HasValue)
+            {
+                var allowedMentions = args.AllowedMentions.Value;
+                if (allowedMentions.AllowedTypes.Value.HasFlag(AllowedMentionTypes.Users)
+                && allowedMentions.UserIds != null && allowedMentions.UserIds.Count > 0)
+                {
+                    throw new ArgumentException("The Users flag is mutually exclusive with the list of User Ids.", nameof(args.AllowedMentions));
+                }
 
+                if (allowedMentions.AllowedTypes.Value.HasFlag(AllowedMentionTypes.Roles)
+                && allowedMentions.RoleIds != null && allowedMentions.RoleIds.Count > 0)
+                {
+                    throw new ArgumentException("The Roles flag is mutually exclusive with the list of Role Ids.", nameof(args.AllowedMentions));
+                }
+            }
+
+            var response = new API.InteractionResponse
+            {
+                Type = InteractionResponseType.UpdateMessage,
+                Data = new API.InteractionCallbackData
+                {
+                    Content = args.Content,
+                    AllowedMentions = args.AllowedMentions.IsSpecified ? args.AllowedMentions.Value?.ToModel() : Optional<API.AllowedMentions>.Unspecified,
+                    Embeds = args.Embeds.IsSpecified ? args.Embeds.Value?.Select(x => x.ToModel()).ToArray() : Optional<API.Embed[]>.Unspecified,
+                    Components = args.Components.IsSpecified
+                        ? args.Components.Value?.Components.Select(x => new API.ActionRowComponent(x)).ToArray()
+                        : Optional<API.ActionRowComponent[]>.Unspecified,
+                    Flags = args.Flags.IsSpecified ? (int?)args.Flags.Value ?? Optional<int>.Unspecified : Optional<int>.Unspecified
+                }
+            };
+
+            await InteractionHelper.SendInteractionResponse(this.Discord, response, this.Id, this.Token, options);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<RestFollowupMessage> FollowupAsync(
+            string text = null,
+            Embed[] embeds = null,
```

Comment:
Since were now in full embed array land, should we move the `embeds` field to the end and add a `params` keyword so you dont need to construct an array?

Reply:
idk what is better 🤷

Reply:
my issue is that you have to create a new array: `new Embed[] { ... }` if its at the end of the function with the params keyword you can just pass in embeds `embeds: new EmbedBuilder(...), new EmbedBuilder(...)`

Reply:
I think its bad, because need specify all arguments for specify embeds without creating array, else you again need create array. I dont think what hard to write `new[] { new EmbedBuilder(...) }`?

Reply:
`embeds: new EmbedBuilder(...), new EmbedBuilder(...)` doesnt work... Maybe i idiot, but i test it now with integers
![image](https://user-images.githubusercontent.com/47792796/127015962-20df0902-ee58-4281-bd81-e27e9e5516cc.png)

Reply:
But, I noticed, if embeds will be with `params`, then we can put `embeds: new EmbedBuilder(...)` or `embeds: new[] { new EmbedBuilder(...), new EmbedBuilder(...) }` 🤔

Reply:
Considering what I noticed in my last comment, I think to do it

Reply:
Ah i see, then its fine the way it is